src/pharmpy/modeling/estimation_steps.py

```diff
@@ -400,3 +400,108 @@ def set_evaluation_step(model: Model, idx: int = -1):
         newsteps = steps[0:-1] + newstep
     model = model.replace(estimation_steps=newsteps)
     return model.update_source()
+
+
+def add_predictions_residuals(model: Model, pred: List[str] = [], res: List[str] = []):
```

Comment:
I would like two different functions: `add_predictions` and `add_residuals`. Also the default values should not be mutable.